engine/player/unique_gear_shadowlands.cpp

```diff
@@ -2335,6 +2335,66 @@ void ebonsoul_vise( special_effect_t& effect )
 
   effect.execute_action = create_proc_action<ebonsoul_vise_t>( "ebonsoul_vise", effect );
 }
+/**Shadowed Orb of Torment
+ * id=355321 driver
+ * id=356326 mastery buff
+ * id=356334 mastery buff value
+ */
+void shadowed_orb_of_torment( special_effect_t& effect )
+{
+  struct tormented_insight_channel_t : public proc_spell_t
+  {
+    buff_t* buff;
+
+    tormented_insight_channel_t( const special_effect_t& e, buff_t* b )
+      : proc_spell_t( "tormented_insight", e.player, e.driver(), e.item ), buff( b )
+    {
+      channeled    = true;
+      hasted_ticks = harmful = false;
+    }
+
+    timespan_t tick_time( const action_state_t* ) const override
+    {
+      return base_tick_time;
+    }
+
+    void execute() override
+    {
+      proc_spell_t::execute();
+
+      event_t::cancel( player->readying );
+      player->reset_auto_attacks( data().duration() );
+    }
+
+    void tick( dot_t* d ) override
+    {
+      buff->trigger( 1, buff_t::DEFAULT_VALUE(), 1.0, buff->buff_duration() );
```

Comment:
This trinket appears to actually apply the buff when the channel completes, based on the number of ticks that occurred. I think you should just be able to reference d->current_tick while in the last_tick method below to get the number of ticks that happened and trigger the buff with the appropriate full duration there.

Reply:
Right, I've missed that. It should behave correctly now.